### PR TITLE
Add configure_additional_bank Functionality and Bump Version to 1.2.17

### PR DESCRIPTION
**Pull request by gmorales96**
### Description

This pull request introduces the following changes:

### New Functionality
- Adds the `add_bank` function to configure an additional bank by updating the `BANKS` and `BANK_NAMES` dictionaries.
- Adds the `remove_bank`function to delete a bank by updating the `BANKS` and `BANK_NAMES` dictionaries.

### Version Update
- Bumps the project version from `1.2.16` to `1.2.17` to reflect the addition of the new functionality.

### Unit Tests
- Added unit tests to ensure:
  - The function successfully updates `BANKS` and `BANK_NAMES` when provided valid inputs.
  - The function raises the correct exceptions.

## Summary by CodeRabbit

- **New Features**
  - Added functionality to configure a new bank within the library, including methods to add and remove banks.
  - Introduced a new data model for bank configuration.

- **Bug Fixes**
  - Enhanced validation for bank codes and names to ensure proper input.

- **Documentation**
  - Updated the README to include a new section on configuring banks.

- **Tests**
  - Added new test cases to validate the functionality of the bank configuration and removal features.

- **Chores**
  - Updated version number to reflect the new release.

### CODE REVIEW

File: tests/test_clabe.py [hunk 1]
```diff
@@ -36,3 +41,31 @@ def test_generate_new_clabes():
     for clabe in clabes:
         assert clabe.startswith(prefix)
         assert validate_clabe(clabe)
+
+
+def test_configure_additional_bank_success():
```
Comment (pachCode) [inline tests/test_clabe.py:46]: puedes agregar estos bank_codes al test https://github.com/cuenca-mx/spei-validations/blob/5fb22cfd7a590ea573c2c5aa3221206b29768a8c/spei_validations/__init__.py#L6
son los de beta

File: tests/test_clabe.py [hunk 2]
```diff
@@ -36,3 +41,31 @@ def test_generate_new_clabes():
...
+    with pytest.raises(BankCodeABMAlreadyExistsError):
+        configure_additional_bank("002", "40002", "Banamex")
+
+
+def test_configure_additional_bank_existing_banxico_code():
+    with pytest.raises(BankCodeBanxicoAlreadyExistsError):
+        configure_additional_bank("666", "40137", "New Bank")
+
+
+def test_configure_additional_bank_invalid_inputs():
```
Comment (pachCode) [inline tests/test_clabe.py:61]: creo que podrías parametrizar este test

File: tests/test_clabe.py [hunk 3]
```diff
@@ -36,3 +37,32 @@ def test_generate_new_clabes():
...
+    'abm_code, banxico_code, name',
+    [
+        ('713', '90713', 'Cuenca DMZ'),
+        ('777', '713', 'Cuenca Gem DMZ'),
+        ('666', '723', 'Cuenca Gem Beta'),
+    ],
+)
+def test_configure_additional_bank_success(abm_code, banxico_code, name):
+    configure_additional_bank(abm_code, banxico_code, name)
+    assert get_bank_name(abm_code) == name
```
Comment (felipao-mx) [inline tests/test_clabe.py:52]: validar que el código de 5 dígitos también esté correctamente configurado

**Review by felipao-mx — CHANGES_REQUESTED** [review]
algunos cambios

File: clabe/validations.py [hunk 4]
```diff
@@ -61,3 +64,52 @@ def generate_new_clabes(number_of_clabes: int, prefix: str) -> List[str]:
...
+        min_length=5, max_length=5, description="The Banxico code for the bank"
+    )
+
+    @validator("bank_code_banxico")
+    def validate_bank_code(cls, value):
+        if not re.fullmatch(r"\d{5}", value):
+            raise ValueError(
+                "bank_code_banxico must be a string of exactly 5 digits"
+            )
+        return value
```
Comment (felipao-mx) [inline clabe/validations.py:86]: no es necesario este validator, se puede hacer con Field. Sugiero:
```python
from pydantic import BaseModel, Field

class Bank(BaseModel):
    bank_code: str = Field(..., regex="^\d{5}$")
    bank_name: str = Field(..., min_length=1)
```

File: clabe/validations.py [hunk 5]
```diff
@@ -61,3 +64,52 @@ def generate_new_clabes(number_of_clabes: int, prefix: str) -> List[str]:
...
+
+    @property
+    def bank_code_abm(self):
+        return self.bank_code_banxico[-3:]
+
+
+def configure_additional_bank(bank_code_banxico: str, bank_name: str) -> None:
```
Comment (felipao-mx) [inline clabe/validations.py:93]: sugiero que se llame solo `add_bank`

File: clabe/validations.py [hunk 6]
```diff
@@ -61,3 +64,52 @@ def generate_new_clabes(number_of_clabes: int, prefix: str) -> List[str]:
...
+            abm
+            for abm, banxico in BANKS.items()
+            if banxico == bank_code_banxico
+        ),
+        None,
+    )
+
+    if bank_code_abm:
+        BANKS.pop(bank_code_abm)
+        BANK_NAMES.pop(bank_code_banxico)
```
Comment (felipao-mx) [inline clabe/validations.py:115]: sugiero 
```python
try:
    del BANK_NAMES[bank_code_banxico]
except KeyError:
    ...
try:
    del BANKS[bank_code_banxico[-3:]]
except KeyError:
    ...
```

File: tests/test_clabe.py [hunk 7]
```diff
@@ -36,3 +38,36 @@ def test_generate_new_clabes():
...
+        ('123AT', 'Test Bank'),  # Non-numeric codes
+    ],
+)
+def test_configure_additional_bank_invalid_inputs(banxico_code, name):
+    with pytest.raises(ValueError):
+        configure_additional_bank(banxico_code, name)
+
+
+def test_remove_bank():
+    remove_bank('40138')
```
Comment (felipao-mx) [inline tests/test_clabe.py:71]: falta assert para asegurar que `BANK` y `BANK_NAMES` ya no contengan el código y nombre que quieres eliminar

**Review by felipao-mx — CHANGES_REQUESTED** [review]

**Review by felipao-mx — APPROVED** [review]

**Review by pachCode — APPROVED** [review]

**Pull request merged**